src/adapter/src/coord.rs

```diff
@@ -2801,10 +2801,57 @@ impl<S: Append + 'static> Coordinator<S> {
             introspection_sources,
         }];
 
-        for (replica_name, config) in replicas {
+        let azs = self.catalog.state().availability_zones();
+        let mut n_replicas_per_az = azs
+            .iter()
+            .map(|s| (s.as_str(), 0))
+            .collect::<HashMap<_, _>>();
+        for (_name, r) in replicas.iter() {
+            if let Some(az) = r.get_az() {
+                let ct: &mut usize = n_replicas_per_az.get_mut(az).ok_or_else(|| {
+                    AdapterError::InvalidClusterReplicaAz {
+                        az: az.to_string(),
+                        expected: azs.to_vec(),
+                    }
+                })?;
+                *ct += 1
+            }
+        }
+        let min = *n_replicas_per_az
+            .values()
+            .min()
+            .expect("Must have at least one availability zone");
+        let argmins = n_replicas_per_az
+            .iter()
+            .filter_map(|(k, v)| (*v == min).then(|| k))
+            .collect::<Vec<_>>();
+
+        let mut i_default_az = 0;
+        for (replica_name, config) in replicas.into_iter() {
+            // If the AZ was not specified, choose one, round-robin, from the ones with
+            // the lowest number of configured replicas for this cluster.
```

Comment:
It doesn't much matter, but I don't think this is quite optimal? Say there are three AZs. If you have one replica constrained to AZ1, one replica constrained to AZ2, and three unconstrained, one this put all three of those in AZ3 instead of smearing them?

Reply:
You're right, thanks for pointing that out! That would have been an embarrassing bug!